test/framework/src/main/java/org/elasticsearch/test/disruption/LongGCDisruptionTest.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.test.disruption;
+
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.regex.Pattern;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThan;
+
+public class LongGCDisruptionTest extends ESTestCase {
+
+    static class LockedExecutor {
+        ReentrantLock lock = new ReentrantLock();
+
+        public void executeLocked(Runnable r) {
+            lock.lock();
+            try {
+                r.run();
+            } finally {
+                lock.unlock();
+            }
+        }
+    }
+
+    public void testBlockingTimeout() throws Exception {
+        final String nodeName = "test_node";
+        LongGCDisruption disruption = new LongGCDisruption(random(), nodeName) {
+            @Override
+            protected Pattern[] getUnsafeClasses() {
+                return new Pattern[]{
+                    Pattern.compile("LockedExecutor")
+                };
+            }
+
+            @Override
+            protected long getStoppingTimeoutInMillis() {
+                return 100;
+            }
+        };
+        final AtomicBoolean stop = new AtomicBoolean();
+        final CountDownLatch underLock = new CountDownLatch(1);
+        final CountDownLatch pauseUnderLock = new CountDownLatch(1);
+        final LockedExecutor lockedExecutor = new LockedExecutor();
+        final AtomicLong ops = new AtomicLong();
```

Comment:
I do not see the point of this variable, it's incremented but otherwise never read?